web/src/search_suggestion.ts

```diff
@@ -729,46 +729,78 @@ function get_operator_suggestions(last: NarrowTerm): Suggestion[] {
     });
 }
 
+// One full search suggestion can include multiple search terms, based on what's
+// in the search bar.
+type SuggestionLine = Suggestion[];
+function suggestion_search_string(suggestion_line: SuggestionLine): string {
+    const search_strings = [];
+    for (const suggestion of suggestion_line) {
+        if (suggestion.search_string !== "") {
+            search_strings.push(suggestion.search_string);
+        }
+    }
+    return search_strings.join(" ");
+}
+
 class Attacher {
-    result: Suggestion[] = [];
+    result: SuggestionLine[] = [];
     prev = new Set<string>();
-    base: Suggestion;
+    base: SuggestionLine;
 
-    constructor(base: Suggestion) {
+    constructor(base: SuggestionLine) {
         this.base = base;
     }
 
-    prepend_base(suggestion: Suggestion): void {
-        if (this.base && this.base.description_html.length > 0) {
-            suggestion.search_string = this.base.search_string + " " + suggestion.search_string;
-            suggestion.description_html =
-                this.base.description_html + ", " + suggestion.description_html;
-        }
-    }
-
-    push(suggestion: Suggestion): void {
-        if (!this.prev.has(suggestion.search_string)) {
-            this.prev.add(suggestion.search_string);
-            this.result.push(suggestion);
+    push(suggestion_line: SuggestionLine): void {
+        const search_string = suggestion_search_string(suggestion_line);
+        if (!this.prev.has(search_string)) {
+            this.prev.add(search_string);
+            this.result.push(suggestion_line);
         }
     }
 
     push_many(suggestions: Suggestion[]): void {
         for (const suggestion of suggestions) {
-            this.push(suggestion);
+            this.push([suggestion]);
         }
     }
```

Comment:
What's the reasoning behind having both `push_many` and `push`? It looks like previously, `push` only accepted a single suggestion, while `push_many` accepted a list, but you seem to be rewriting `push` to accept a list anyway, yet with different implementations for the two.

Reply:
`push_many` pushes multiple suggestion lines, and `push` only pushes one.

Maybe a clearer way to write this is `push_many(suggestion_lines: SuggestionLine[])` where the caller converts the suggestions to suggestion lines instead of `push_many`. Would that help?